src/win/pipe.c

```diff
@@ -1687,41 +1709,69 @@ static DWORD uv__pipe_read_ipc(uv_loop_t* loop, uv_pipe_t* handle) {
   } else {
     /* Start of a new IPC frame. */
     uv__ipc_frame_header_t frame_header;
+    uint32_t xfer_flags;
+    uv__ipc_socket_xfer_type_t xfer_type;
     uv__ipc_socket_xfer_info_t xfer_info;
 
     /* Read the IPC frame header. */
+    assert(sizeof frame_header == 16);
```

Comment:
Are this and the other size assertion about guaranteeing compatibility? If so, maybe add comments - otherwise they look oddly specific.

Reply:
Detecting incompatible protocol changes -- that's indeed the purpose.
I've turned those asserts into STATIC_ASSERT, moved them to the top of the file, and added a comment.
https://github.com/libuv/libuv/blob/27ba66281199bdcade823677af8dedc161152fb6/src/win/pipe.c#L81-L84